### PR TITLE
Refactor load spinner/message

### PR DESCRIPTION
**Pull request by p3rcypj**
### :pushpin: References

-   **Issue:** Closes https://app.clickup.com/t/865cju10n

### :memo: Implementation

Added a multi-message loader prioritized because the right panel was left blank in the case when the pdb data info request wasn't already complete and the 3dviewer already finished.

### :art: Screenshots

image

### :fire: Testing

- 6zow+EMD-11328 (I saw the "Starting 3D Viewer..." message but I am testing it connected to my phone network, so a lower internet speed may affect how long I saw the message)
- 6aba (the molstar requests finish before the "Load PDB Data..." so this message appears 2 times now, that was the initial bug that we wanted to solve because it was showing the right panel blank without a message)

### CODE REVIEW

**Review by tokland — APPROVED** [review]
The idea looks good to me! That's something that we've always wanted for the snackbar in d2-ui-compoennt, some multi-message prioritized notification. Let's see how it work here.

**Comment by p3rcypj** [comment]
@tokland The loader with key `updateVisualPlugin` is not being loaded any time (MolecularStructure#L121) so the line is never executed. Could you take a lot at it?

**Comment by p3rcypj** [comment]
> The idea looks good to me! That's something that we've always wanted for the snackbar in d2-ui-component, some multi-message prioritized notification. Let's see how it work here.

Thanks @tokland!! 🎉😊

File: app/assets/javascripts/3dbio_viewer/src/webapp/components/molecular-structure/MolecularStructure.tsx [hunk 1]
```diff
@@ -44,12 +44,8 @@ interface MolecularStructureProps {
...
     onLigandsLoaded(ligands: Ligand[]): void;
     proteinNetwork: Maybe<ProteinNetwork>;
-    title: string;
-    setTitle: (title: string) => void;
-    isLoading: boolean;
-    showLoading: () => void;
-    hideLoading: () => void;
-    setError: (message: string) => void;
+    loaderBusy: boolean;
+    updateLoaderOnResolve: <T>(key: LoaderKey, promise: Promise<T>, message?: string) => Promise<T>;
```
[the remark this answers is not in the document]
Comment (p3rcypj) [reply app/assets/javascripts/3dbio_viewer/src/webapp/components/molecular-structure/MolecularStructure.tsx:48]: Also on the hook?

**Pull request merged**